src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -168,10 +169,11 @@ public override BoundNode VisitITuplePattern(BoundITuplePattern node)
                     }
                     break;
                 case BoundNegatedPattern p:
-                    // PROTOTYPE(ngafter): what is the right behavior here?
+                    LearnFromAnyNullPatterns(inputSlot, inputType, p.Negated);
                     break;
                 case BoundBinaryPattern p:
-                    // PROTOTYPE(ngafter): what is the right behavior here?
+                    LearnFromAnyNullPatterns(inputSlot, inputType, p.Left);
```

Comment:
>LearnFromAnyNullPatterns(inputSlot, inputType, p.Left); [](start = 20, length = 55)

Should we take into account where we are dealing with an `or` than an `and` here?

Reply:
No.  A pure null test under either permits us to infer that the input may have been null.

---
In reply to: [381009931](https://github.com/dotnet/roslyn/pull/41503#discussion_r381009931) [](ancestors = 381009931)